gluejobutils/s3.py

```diff
@@ -1,21 +1,130 @@
-from gluejobutils.datatypes import translate_metadata_type_to_type, get_customschema_from_metadata
 import json
 import boto3
+import botocore
+
+from gluejobutils.utils import add_slash, remove_slash
+
+try :
+    # python 2.7
+    from StringIO import StringIO
+except :
+    # python 3.6
+    from io import StringIO
+
 s3_resource = boto3.resource('s3')
 
-def s3_path_to_bucket_key(path):
-    path = path.replace("s3://", "")
-    bucket, key = path.split('/', 1)
-    return bucket, key
+def bucket_key_to_s3_path(bucket, key) :
+    """
+    Takes an S3 bucket and key combination and returns the full S3 path to that location.
+    """
+    return 's3://{}/{}'.format(bucket, key)
 
-def spark_read_csv_using_metadata_path(spark, metadata_path, csv_path, **kwargs):
-    meta_json = read_json_from_s3(metadata_path)
-    schema = get_customschema_from_metadata(meta_json)
-    df = spark.read.csv(csv_path, schema=schema, **kwargs)
-    return df
+def s3_path_to_bucket_key(s3_path):
+    """
+    Splits out s3 file path to bucket key combination
+    """
+    s3_path = s3_path.replace("s3://", "")
+    bucket, key = s3_path.split('/', 1)
+    return bucket, key
 
-def read_json_from_s3(s3_path):
+def read_json_from_s3(s3_path, encoding = 'utf-8') :
+    """
+    read a json file from an s3 path
+    """
     bucket, key = s3_path_to_bucket_key(s3_path)
     obj = s3_resource.Object(bucket, key)
-    text = obj.get()['Body'].read().decode('utf-8')
-    return json.loads(text)
\ No newline at end of file
+    text = obj.get()['Body'].read().decode(encoding)
+    return json.loads(text)
+
+def write_json_to_s3(data, s3_path) :
+    """
+    Saves your data to a json file (in memory and then sends it to the s3 path provided)
+    """
+    bucket, key = s3_path_to_bucket_key(s3_path)
+    log_file = StringIO()
+    json.dump(data, log_file, indent=4, separators=(',', ': '))
+    log_obj = s3_resource.Object(bucket, key)
+    log_upload_resp = log_obj.put(Body=log_file.getvalue())
+    return log_upload_resp
+
+def get_filepaths_from_s3_folder(s3_folder_path, parquet = False) :
```

Comment:
Yeah I am not sure about this function in general - because it will return true if a parquet exists anywhere with a prefix.
i.e. 
```
parquet_file_at = 's3://bucket/data/some/longer/path/file.parquet'
get_filepaths_from_s3_folder('s3://bucket/data', parquet = True) # True 
```

Only reason why I have left it in is that it is useful. I don't want to ignore any filepaths with deeper folders either as it wouldn't detect parquet files with HIVE partitions.

Reply:
Agreed